scripts/build-engine/tests/constant-manager/constant-manager.test.js

```diff
@@ -0,0 +1,77 @@
+const { ConstantManager } = require('../../dist/constant-manager/index');
+const fs = require('fs');
+const ps = require('path');
+
+const cm = new ConstantManager(__dirname);
+
+// The different OS handle escape like `\n` and `\t` differently, so we need to smooth this difference.
+function handleEscape(str) {
+    str = str.replace(/\s/g, '');
+    return str;
+}
+
+test('generateInternalConstants', () => {
+    const result = handleEscape( cm.generateInternalConstants() );
+    const targetContent = handleEscape( fs.readFileSync(ps.join(__dirname, 'internal-constants.txt'), 'utf8') );
+    expect(result).toBe(targetContent);
```

Comment:
`expect(result).toMatchSnapshot()` would be more convenient: you then don't need read the snapshot and handle escaping yourself.

Reply:
great idea